core-java-modules/core-java-concurrency-2/src/test/java/com/baeldung/concurrent/completablefuture/CombiningCompletableFuturesUnitTest.java

```diff
@@ -0,0 +1,105 @@
+package com.baeldung.concurrent.completablefuture;
+
+import static java.util.function.Predicate.isEqual;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+
+public class CombiningCompletableFuturesUnitTest {
+
+    @Mock Logger logger;
+
+    @BeforeEach
+    void setup() {
+        logger = mock(Logger.class);
+    }

```

Comment:
The purpose of the `@Mock` annotation is to use it in conjunction with `@ExtendWith(MockitoExtension.class`

Either use `@ExtendWith`, or just initialize it directly. You also don't need a `@BeforeEach` for something like this - a new instance of the test class is created on each execution of a test.
```suggestion
     private final Logger logger = mock(Logger.class);
```